src/linear_algebra/affine_space/affine_subspace.lean

```diff
@@ -1119,6 +1119,43 @@ instance {s : set P} [nonempty s] : nonempty (affine_span k s) :=
 
 variables {k}
 
+/--
+An induction principle for span membership. If `p` holds for all elements of `s` and is
+preserved under certain affine combinations, then `p` holds for all elements of the span of `s`.
+-/
+lemma affine_span_induction {x : P} {s : set P} {p : P → Prop} (h : x ∈ affine_span k s)
+  (Hs : ∀ x : P, x ∈ s → p x)
+  (Hc : ∀ (c : k) (u v w : P), p u → p v → p w → p (c • (u -ᵥ v) +ᵥ w)) : p x :=
+(@affine_span_le _ _ _ _ _ _ _ _ ⟨p, Hc⟩).mpr Hs h
+
+/-- A dependent version of `affine_span_induction`. -/
+lemma affine_span_induction' {x : P} {s : set P} {p : Π x, x ∈ affine_span k s → Prop}
+  (h : x ∈ affine_span k s)
```

Comment:
My memory with `submodule.span_induction'` is that this behaved better when `x` and `h` were the very last arguments; am I imagining that the order is different there?

Reply:
That's right, I'll change the order.